provider/inspec.rb

```diff
@@ -73,8 +74,81 @@ def generate_typed_array(data, prop) end
 
     def emit_resourceref_object(data) end
 
-    def emit_nested_object(data) end
-
     def generate_network_datas(data, object) end
+
+    def emit_nested_object(data)
+      target = if data[:emit_array]
+                 data[:property].item_type.property_file
+               else
+                 data[:property].property_file
+               end
+      {
+        source: File.join('templates', 'inspec', 'nested_object.erb'),
+        target: "libraries/#{target}.rb",
+        overrides: emit_nested_object_overrides(data)
+      }
+    end
+
+    def emit_nested_object_overrides(data)
+      data.clone.merge(
+        api_name: data[:api_name].camelize(:upper),
+        object_type: data[:obj_name].camelize(:upper),
+        product_ns: data[:product_name].camelize(:upper),
+        class_name: if data[:emit_array]
+                      data[:property].item_type.property_class.last
+                    else
+                      data[:property].property_class.last
+                    end
+      )
+    end
+
+    def primitive?(property)
+      is_primitive = property.is_a?(::Api::Type::Primitive)
+      is_primitive_array = (property.is_a?(Api::Type::Array)\
+        && !property.item_type.is_a?(::Api::Type::NestedObject))
+      is_primitive || is_primitive_array
+    end
+
+    def resource_ref?(property)
+      property.is_a?(::Api::Type::ResourceRef)
+    end
+
+    def typed_array?(property)
+      property.is_a?(::Api::Type::Array)
+    end
+
+    def nested_object?(property)
+      property.is_a?(::Api::Type::NestedObject)
+    end
+
+    def generate_requires(properties, requires = [])
+      nested_props = properties.select { |type| nested_object?(type) }
+      requires.concat(properties.reject { |type| no_requires?(type) }\
+        .collect { |type| easy_requires(type) })
+      requires.concat(nested_props.map { |nested_prop| generate_requires(nested_prop.properties) })
+      requires.concat(nested_props.map(&:property_file))
```

Comment:
Can you move a copy of `property_file` over to `provider/inspec.rb`? It'll help make the transition easier down the line when we separate Puppet + Chef out